### PR TITLE
Swap out the Initialize method for proper constructors

### PR DESCRIPTION
**Pull request by christopher-henderson**
This PR addresses #606 (where I have much of the impetus and writeup for work in question).

Broadly speaking, to accomplish this, this change implements...

1. Swaps out the `Initialize` method from `LintInterface`
2. Changes `Lint.Lint` to be a function that returns a `LintInterface` rather than a `LintInterface` itself.
3. Removed all of the wiring and testing around `Initialize`
4. Updated the template script to auto generate a constructor.
5. Applied a find-and-replace on every lint to migrate them over to constructors.

For the sake of sharing the joy/pain, here's the regexes in question fed through Goland

```
Match: func\(\) lint.LintInterface \{ return &([a-zA-Z])(.*)\{\} \},\n(\s*)\}\)\n\}\n\n
Replace: New\u$1$2,\n$3\}\)\n\}\n\nfunc New\u$1$2() lint.LintInterface{\n\treturn &$1$2{}\n}\n\n
```

All of the lints appear to be fully functional, although I can imagine there be formatting that might be off-kilter after this find and replace (like a comment that is left orphaned or is out of order now).

### CODE REVIEW

File: v3/lint/registration.go [hunk 1]
```diff
@@ -127,25 +127,12 @@ func (e errDuplicateName) Error() string {
 		e.lintName)
 }
 
-// errBadInit is returned from registry.Register if the provided lint's
```
Comment (christopher-henderson) [inline v3/lint/registration.go:130]: No longer needed this aspect of the test.

File: v3/lints/apple/lint_ct_sct_policy_count_unsatisfied.go [hunk 2]
```diff
@@ -33,15 +33,16 @@ func init() {
 		Citation:      "https://support.apple.com/en-us/HT205280",
```
Comment (christopher-henderson) [inline v3/lints/apple/lint_ct_sct_policy_count_unsatisfied.go:33]: From here on out it is (thankfully) pretty repetitive. It makes a new constructor for every type, deletes `Initialize`, and replaces `Lint: &structName{}` with `Lint: NewStructName`.

File: v3/newLint.sh [hunk 3]
```diff
@@ -44,6 +44,7 @@ FILENAME=${LINTNAME:2}
 STRUCTNAME=$3
 
 sed -e "s/PACKAGE/${PATHNAME}/" \
+    -e "s/PASCAL_CASE_SUBST/${STRUCTNAME^}/g" \
```
Comment (christopher-henderson) [inline v3/newLint.sh:47]: Today I learned that `{$varname^}` will uppercase the first character in the string. Bash is endless.

**Review by dadrian — APPROVED** [review]
Apologies for the delay...went through this when you first posted it but didn't click approve.

Good work!

**Review by cpu — APPROVED** [review]
This makes sense to me! Thanks for all of the context in the PR desc/issue.

**Comment by christopher-henderson** [comment]
Thank you folks! I was eyeing the merge button over the weekend, but then a pinch of food poisoning kicked me in the teeth. I appreciate ya'll driving it to completion.

**Pull request merged**